### ISSUE TITLE
Support for patching over multiple datasets in Zarrs

### ISSUE THREAD

**Issue by Mohinta2892** [BiaPyX/BiaPy#82]
Hello Dani!

Please can you extend the support for loading and patching from multiple datasets (such as labels and raw) stored within `one` zarr.

The zarr organisation may look like:

```bash
 filename.zarr
       - volumes
              - raw
              - labels
                   - neuron_ids
```

The datasets may also have associated attributes, such as offsets (for labels) and image resolution ( for raw).

An example can be found [here](https://www.dropbox.com/scl/fo/856ztdswex3ywu95xh7qb/ABdNQ5xG68TpZv35TecgNK4?rlkey=t463sx2vc92bh6eho4y0gih0b&st=qxc75c16&dl=0)

Best,
Samia

**Comment by danifranco** [comment]
Hello!

Just done in [e0dfa47](https://github.com/BiaPyX/BiaPy/commit/e0dfa47d2e7ff886f04aa6a627f481fe766b2895). Feel free to close the issue.

Best,
Dani

**Comment by danifranco** [comment]
I'm closing this Samia. Feel free to reopen again te issue if you want to add something else.

**Comment by Mohinta2892** [comment]
Hi Dani,

Reopening this since it is throwing an error:

```
[16:03:25.064177] Checking ground truth classes in /{path}/sample_A_20160501.zarr . . .
  0%|                                                                                                                                       | 0/1 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/samia/anaconda3/envs/BiaPy_env/bin/biapy", line 8, in <module>
    sys.exit(main())
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/biapy/__init__.py", line 67, in main
    _biapy = BiaPy(**vars(args))
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/biapy/_biapy.py", line 201, in __init__
    self.workflow = getattr(mdl, name)(self.cfg, self.job_identifier, self.device, self.args)
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/biapy/engine/semantic_seg.py", line 43, in __init__
    check_masks(cfg.DATA.TRAIN.GT_PATH, n_classes=cfg.MODEL.N_CLASSES, is_3d=(self.cfg.PROBLEM.NDIM == "3D"))
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/biapy/data/data_manipulation.py", line 3000, in check_masks
    img = read_img_as_ndarray(os.path.join(path, ids[i]), is_3d=is_3d)
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/biapy/data/data_manipulation.py", line 2925, in read_img_as_ndarray
    img = imread(path)
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/skimage/io/_io.py", line 60, in imread
    img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/skimage/io/manage_plugins.py", line 217, in call_plugin
    return func(*args, **kwargs)
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/skimage/io/_plugins/imageio_plugin.py", line 11, in imread
    out = np.asarray(imageio_imread(*args, **kwargs))
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/imageio/v3.py", line 53, in imread
    with imopen(uri, "r", **plugin_kwargs) as img_file:
  File "/home/samia/anaconda3/envs/BiaPy_env/lib/python3.10/site-packages/imageio/core/imopen.py", line 281, in imopen
    raise err_type(err_msg)
OSError: Could not find a backend to open /{path}/sample_A_20160501.zarr/.zgroup`` with iomode `r`.
```
Not sure, why it is attempting to open the .zgroup.

I have set these in the config:
```
        INPUT_ZARR_MULTIPLE_DATA_RAW_PATH: volumes.raw
        INPUT_ZARR_MULTIPLE_DATA_GT_PATH: volumes.labels.neuron_ids
        INPUT_ZARR_MULTIPLE_DATA: True
```

Best,
Samia

**Comment by danifranco** [comment]
Solved with 46e2b7498e4b67afc9efaf5693d4023d74aec8d6 and afa534b437ec04d4a03965358a68f629c2f94b28 . Tell me if you still have issues please!
